pkg/kvstore/etcd.go

```diff
@@ -852,15 +852,16 @@ reList:
 			if traceEnabled {
 				scopedLog.Debugf("Emitting EventTypeDelete event for %s", k)
 			}
-
-			queueStart := spanstat.Start()
-			w.Events <- event
-			trackEventQueued(scope, EventTypeDelete, queueStart.End(true).Total())
+			if ok := w.emit(ctx, scope, event); !ok {
+				return
+			}
 		})
 
 		// Only send the list signal once
 		if !listSignalSent {
-			w.Events <- KeyValueEvent{Typ: EventTypeListDone}
+			if ok := w.emit(ctx, scope, KeyValueEvent{Typ: EventTypeListDone}); !ok {
+				return
+			}
```

Comment:
Nit. Could be shortened as follows (same for the other occurrences):

```suggestion
			if !w.emit(ctx, scope, KeyValueEvent{Typ: EventTypeListDone}) {
				return
			}
```